provider/bigquery/bigquery.go

```diff
@@ -146,24 +146,26 @@ func (c *bigQueryClient) RevokeDatasetAccess(ctx context.Context, d *Dataset, us
 	if err != nil {
 		return err
 	}
-
 	bqRole, err := c.ResolveDatasetRole(role)
 	if err != nil {
 		return err
 	}
-	var removeIndex int
-	for i, a := range metadata.Access {
+
+	isAccessFound := false
+	newAccessEntries := []*bq.AccessEntry{}
+	for _, a := range metadata.Access {
 		if a.Entity == user && a.Role == bqRole {
-			removeIndex = i
-			break
+			isAccessFound = true
+			continue
 		}
+		newAccessEntries = append(newAccessEntries, a)
 	}
-	if removeIndex == 0 {
+	if !isAccessFound {
```

Comment:
instead of using !isAccessFound can we use this instead
```
// same length means no changes
if len(newAccessEntries) == len(metadata.Access) {
  return ErrPermissionNotFound
}
```

Reply:
Yes, avoiding extra variables. Changed as suggested.